tardis/__init__.py

```diff
@@ -33,6 +33,126 @@
+        filter() allows to set the logging level for
+        all the record that are being parsed & hence remove those
+        which are not of the particular level
+
+        Parameters
+        ----------
+        log_record : logging.record
+            which the paricular record upon which the
+            filter will be applied
+
+        Returns
+        -------
+        boolean : True, if the current log_record has the
+            level that of the specified log_level
+            False, if the current log_record doesn't have the
+            same log_level as the specified one
+        """
+        return log_record.levelno == self.log_level
+
+
+# Setting up a list to store the Logging Filters set by logger.addFilter()
+list_of_filter = []
+
+
+def logging_state(log_state, tardis_config, specific):
+    """
+    Function to set the logging configuration for the simulation output
+    Called from within run_tardis()
+    Configured via functional arguments passed through run_tardis() - log_state & specific
+    Configured via YAML parameters under `debug` section - logging_level & specific_logging
+
+    Parameters
+    ----------
+    log_state: str
+        Allows to input the log level for the simulation
+        Uses Python logging framework to determine the messages that will be output
+    specific: boolean
+        Allows to set specific logging levels. Logs of the `log_state` level would be output.
+    """
+    if "debug" in tardis_config:
+        if specific or tardis_config["debug"]["specific"]:
+            specific = True
+        else:
+            specific = False
+
+        if tardis_config["debug"]["log_state"] or log_state:
+            if (
+                log_state.upper() == "CRITICAL"
+                and tardis_config["debug"]["log_state"]
+            ):
+                logging_level = tardis_config["debug"]["log_state"]
+            elif log_state:
+                logging_level = log_state
+                if tardis_config["debug"]["log_state"] and log_state:
+                    print(
+                        "log_state is defined both in Functional Argument & YAML Configuration {debug section}"
+                    )
+                    print(
+                        f"log_state = {log_state} will be used for Log Level Determination\n"
+                    )
+            else:
+                logging_level = tardis_config["debug"]["log_state"]
+    else:
+        logging_level = log_state
+        specific = specific
+
+    loggers = [
+        logging.getLogger(name) for name in logging.root.manager.loggerDict
+    ]
+    if logging_level.upper() in LOGGING_LEVELS.keys():
+        for logger in loggers:
+            logger.setLevel(LOGGING_LEVELS[logging_level.upper()])
+    else:
+        raise ValueError(
+            f"Passed Value for log_state = {logging_level.upper()} is Invalid. Must be one of the following {list(LOGGING_LEVELS.keys())}"
+        )
+
+    if len(list_of_filter) > 0:
```

Comment:
This is unnecessary since iterating over an empty list won't run anything inside the loop

Reply:
This list gets populated when we add new filters for logging at a particular level. Thus, this snippet of the code is actually checking that we cannot stack multiple filters such as a case where we are filtering `INFO` based log messages inside `DEBUG` log messages. This removes all the filters before applying a new filter.